### ISSUE TITLE
Edit page link in the admin bar and homepage template.

### ISSUE THREAD

**Issue by unprintedch** [WordPress/gutenberg#66109]
### Description

There is a mislieading link to edit the front page from the admin bar.
If you have a page with some content and a template for frontpage setup in the theme editor.
See more ine the step by step

### Step-by-step reproduction instructions

Create a blank playground

Set the sample page to home page via settings > reading
It displays the content of the sample page on the home page.

Create a front-page template in the theme editor.
It displays the content of the front-page template on the home page.

This is understandable, but confusing. one page, two content...

Now if you are on the frontend of your site, click on "edit page" you are sent to edit the page with the sample page content.
And if you hit edit site you go to actual front page template.

Same if you edit the page navigating PAGE > sample page you fine yourself with the sample page content
But if you navigate through APPEARANCE > EDITOR > PAGE > sample page you have the front page content.

It is uncomfortable. We have a single source of truth problem.

### Environment info

It is a basic playground

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

**Comment by unprintedch** [comment]
The ways of the site editors are very strange.
../wp-admin/post.php?post=41147&action=edit
Say template is "pages"
.../wp-admin/site-editor.php?postId=41147&postType=page&canvas=edit
Say template is homepage.
But it is the same page.

Is it me or that is super confusing?

**Comment by unprintedch** [comment]
Anyone with that same problem?

**Comment by Mayank-Tripathi32** [comment]
I agree that this can be confusing. Editing the page content in one place and the front page template in another might feel unclear at first. However, I personally think both serve different purposes: one is focused on defining the website's look and feel, while the other is responsible for the content you want to display on the page.
